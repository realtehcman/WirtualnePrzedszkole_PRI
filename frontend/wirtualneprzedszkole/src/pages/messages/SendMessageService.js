import axios from 'axios'


const SENDMESSAGE_REST_API_URL = 'http://localhost:8080/api/message/'

class SendMessageService {

    d(){
        return axios.post(SENDMESSAGE_REST_API_URL)
    }

    getMessage(id){
        return axios.get(SENDMESSAGE_REST_API_URL)
    }

    SendMessage(message){
        return axios.post(SENDMESSAGE_REST_API_URL, message, {
            headers: {
                'Content-Type': 'application/json'
            }
        });
    }

    SendMessageParents(message){
        return axios.post(SENDMESSAGE_REST_API_URL + 'to_parents', message, {
            headers: {
                'Content-Type': 'application/json'
            }
        });
    }

    SendMessageClasses(id, subject, content){
        return axios.post(SENDMESSAGE_REST_API_URL + 'to_class/' + id, {
            subject: subject,
            content: content,
            headers: {
                'Content-Type': 'application/json'
            }
        });
    }



}

export default new SendMessageService();
