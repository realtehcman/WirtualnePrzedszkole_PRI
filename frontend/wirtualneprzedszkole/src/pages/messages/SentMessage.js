import React from 'react';
import {useNavigate, useParams} from "react-router-dom";
import Sidebar from "../../components/sidebar/sidebar";
import Navbar from "../../components/navbar/navbar";
import {useEffect, useState} from "react";
import MessageService from "./MessageService";
import Popup from "../GroupDisplay/Popup";
import "../GroupDisplay/Popup.css"
import "../User/Table.scss";
import UserService from "../User/UserService";
import SentMessageService from "./SentMessageService";

const Navi2 = (props) => {
    const navigate = useNavigate();
    return (
        <button
            onClick={() => navigate("/StatusMessage/" + props.value, { replace: true })}
            className="btn btn-info"
        >
            Status
        </button>

    );
};

const Navi = (props) => {
    const navigate = useNavigate();
    return (
        <button
            onClick={() => navigate("/ReadMessage/" + props.value, { replace: true })}
            className="btn btn-info"
        >
            Wyświetl
        </button>

    );
};

class SentMessage extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            sent_messages: [],
        };
        this.deleteSentMessages = this.deleteSentMessages.bind(this);
    }

    loger() {
        console.log(this.state);
    }

    deleteSentMessages(id) {
        SentMessageService.deleteSentMessages(id).then((response) => {
            this.setState({
                sent_messages: this.state.sent_messages.filter((sent_messages) => sent_messages.id !== id),
            });
        });
    }

    componentDidMount() {
        SentMessageService.getSentMessages().then((response) => {
            this.setState({ sent_messages: response.data });
            this.loger();
        });
    }




    render() {

        return (
            <div className="scrollable-div">
                <table className="content-table">
                    <thead>
                    <tr className="table-head">
                        <td>Id</td>
                        <td>autor:</td>
                        <td>Temat:</td>
                        <td>Treść</td>
                        <td>Akcje</td>
                    </tr>
                    </thead>
                    <tbody>
                    {this.state.sent_messages.map((sent_messages) => (
                        <tr key={sent_messages.id}>
                            <td>{sent_messages.id}</td>
                            <td>{sent_messages.author}</td>
                            <td>{sent_messages.subject}</td>
                        <td><Navi value={sent_messages.id} />
                            <Navi2 value={sent_messages.id} />
                        </td>
                            <td className="foobar">
                                <button2
                                    onClick={() => this.deleteSentMessages(sent_messages.id)}
                                    className="btn btn-danger"
                                >
                                    Usuń
                                </button2>
                            </td>
                        </tr>
                    ))}
                    </tbody>
                </table>
            </div>

        );
    }
}
export default SentMessage