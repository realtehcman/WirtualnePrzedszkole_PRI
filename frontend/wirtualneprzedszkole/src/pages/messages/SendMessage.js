import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import "../User/Table.scss";
import SendMessageService from "./SendMessageService";
import React, { Component } from "react";
import "../CreateUser/CreateUser.scss";
import UserService from "../User/UserService";
import "./Message.scss";
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.snow.css';


class SendMessage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            to: "",
            subject: "",
            content: "",
            messageSent: false,
            error: false,
            users: [
                {
                    email: "",
                    name: "",
                    id: "",
                },
            ],
            selectedUsers: [],
            filteredUsers: []
        };



        this.changeNameHandler = this.changeNameHandler.bind(this);
        this.changeDescriptionHandler = this.changeDescriptionHandler.bind(this);
        this.GetRecieverHandler = this.GetRecieverHandler.bind(this);
        this.saveMessage = this.saveMessage.bind(this);
        this.saveMessage2 = this.saveMessage2.bind(this);
    }




    componentDidMount() {
        UserService.getUsers().then((response) => {
            this.setState({ users: response.data });
        });
    }


    GetRecieverHandler = (event) => {
        this.setState({ to: event.target.value });
    };


    saveMessage = (e) => {
        e.preventDefault();
        const to = this.state.to.split(',').map(item => item.trim());
        let message = JSON.stringify({
            to: to,
            subject: this.state.subject,
            content: this.state.content,
        });

        SendMessageService.SendMessage(message).then((response) => {
            if (response.data != null) {
                toast.success("wiadomość została wysłana pomyślnie", {
                    position: toast.POSITION.TOP_CENTER,
                });
            } else {
                toast.error("Wystąpił błąd podczas wysyłania wiadomości ", {
                    position: toast.POSITION.TOP_CENTER,
                });
            }
        })
            .catch((error) => {
                toast.error("Wystąpił błąd podczas wysyłania wiadomości", {
                    position: toast.POSITION.TOP_CENTER,
                });
            });
    };




    changeNameHandler = (event) => {
        this.setState({ subject: event.target.value });
    };

    changeDescriptionHandler = (value) => {
        this.setState({ content: value });
    };

    onUserSelect = (user) => {
        this.setState({
            to: user.name + ' ' + user.lastName,
            selectedUsers: [...this.state.selectedUsers, user]
        });
    }

    saveMessage2 = (e) => {
        e.preventDefault();
        let message = JSON.stringify({
            subject: this.state.subject,
            content: this.state.content

        });
        SendMessageService.SendMessageParents(message).then((response) => {
            if (response.data != null) {
                toast.success("wiadomość została wysłana pomyślnie do wszystkich użytkowników", {
                    position: toast.POSITION.TOP_CENTER,
                });
            } else {
                toast.error("Wystąpił błąd podczas wysyłania wiadomości", {
                    position: toast.POSITION.TOP_CENTER,
                });
            }
        })
            .catch((error) => {
                toast.error("Wystąpił błąd podczas wysyłania wiadomości", {
                    position: toast.POSITION.TOP_CENTER,
                });
            });
    };

    render() {

        const {t} = this.props

        const { to, users } = this.state;

        const filteredUsers = users.filter((user) => {
            if (to.includes(user.name + ' ' + user.lastName)) {
                return false;
            }
            const names = to.split(',').map((name) => name.trim());
            const [firstName, lastName] = names[names.length - 1].split(' ');
            return (
                (firstName && user.name.startsWith(firstName)) ||
                ((firstName && user.name.startsWith(firstName)) &&
                    (lastName && user.lastName.startsWith(lastName)))
            );
        });


        return (
            <div>

                <ToastContainer />
                <div className="App_card">
                    <h1>{t('create_a_message')}</h1>
                </div>


                <form className="App_card" onSubmit={(e) => {
                    this.saveMessage(e);

                }}>
                    <div className="mb-3">
                        <label className="form-label font16" htmlFor="name">
                        {t('down')}
                        </label>

                        <input className="form-control" type="PrettyPrintJson" autoComplete="off" id="name" value={to} onChange={this.GetRecieverHandler} />
                        {to.length > 0 && filteredUsers.length > 0 && (
                            <div className="autocomplete-options">
                                {filteredUsers.map((user, index) => (
                                    <div
                                        key={user.id}
                                        className={`autocomplete-option ${index === this.state.selectedOptionIndex ? 'selected' : ''}`}
                                        onClick={() => {
                                            const names = to.split(',').map((name) => name.trim());
                                            names[names.length - 1] = user.name + ' ' + user.lastName;
                                            this.setState({ to: names.join(',') });
                                        }}
                                    >
                                        <div key={user.id} title={user.email}>
                                            {user.name + ' ' + user.lastName}
                                        </div>
                                    </div>
                                ))}
                            </div>
                        )}

                    </div>
                    <div className="mb-3">
                        <label className="form-label font16" htmlFor="email">
                        {t('topic')}
                        </label>
                        <input className="form-control" type="PrettyPrintJson" id="" value={this.state.subject}
                            onChange={this.changeNameHandler} />
                    </div>
                    <div className="mb-3">
                        <label className="form-label font16">
                        {t('contents')}
                        </label>
                        <div className="q1-editor"><ReactQuill value={this.state.content} onChange={this.changeDescriptionHandler} /></div>
                    </div>
                    <div className="form-but d-md-flex align-items-center justify-content-center">

                        <button className="button btn w-auto me-3">{t('send')}</button>
                        <button className="button btn w-auto" onClick={(e) => {
                            this.saveMessage2(e);
                        }}>{t('send_to_all_users')}</button>
                    </div>

                </form>


            </div>
        );
    }
}

export default SendMessage
