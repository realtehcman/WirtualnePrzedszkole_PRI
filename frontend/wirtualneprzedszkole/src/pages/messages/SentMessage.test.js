import React from 'react';
import {configure, shallow} from 'enzyme';
import SendMessage from './SendMessage';
import SendMessageService from './SendMessageService';
import Adapter from "enzyme-adapter-react-16";
import {render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";
import SentMessage from "./SentMessage";

configure({adapter: new Adapter()});

jest.mock('./SendMessageService');

describe('SendMessage component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SendMessage />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const {getByTestId} = render(
        <MemoryRouter>
          <SentMessage/>
        </MemoryRouter>
    );
    expect(getByTestId('sent-message')).toBeTruthy();
  });


});
