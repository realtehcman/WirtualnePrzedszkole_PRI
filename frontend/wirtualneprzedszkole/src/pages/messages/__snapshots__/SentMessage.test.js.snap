// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMessage component should render correctly 1`] = `ShallowWrapper {}`;
